src/Boot/src/Memory.php

```diff
@@ -36,16 +36,27 @@ public function loadData(string $section, string &$filename = null): mixed
 
         try {
             $fp = \fopen($filename, 'r');
+            if ($fp === false) {
+                return null;
+            }
+
             if (!\flock($fp, \LOCK_SH | \LOCK_NB)) {
+                \fclose($fp);
                 return null;
             }
+
             $data = include($filename);
+
             \flock($fp, \LOCK_UN);
-            \fclose($fp);
-            return $data;
         } catch (\Throwable) {
             return null;
+        } finally {
+            if (isset($fp)) {
+                \fclose($fp);
```

Comment:
Looks like `$fp` might be `false` there

Reply:
Here's a Psalm report from the previous run with a check: 

https://github.com/spiral/framework/actions/runs/11093746018/job/30821292030